aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu

```diff
@@ -433,9 +454,19 @@ SparseTensor& mul_out_sparse_cuda(SparseTensor& r_, const SparseTensor& t_, cons
   LongTensor s_indices_ = src._indices();
   Tensor s_values_ = src._values();
   LongTensor r_indices_ = at::empty({sparse_dim, max_nnz}, t_indices_.options());
-  Tensor r_values_ = new_values_with_size_of(t_values_, max_nnz).zero_();
   r_.resize_as_(src);
-  get_sparse_impl(r_)->set_indices_and_values_unsafe(r_indices_, r_values_);
+
+  auto promotedType = promoteTypes(t.scalar_type(), src.scalar_type());
+  TORCH_CHECK(canCast(promotedType, r_.scalar_type()), "Can't convert result type ", promotedType, " to output ", r_.scalar_type());
+
+  if (s_values_.scalar_type() != promotedType) {
```

Comment:
err..what happens if you did all of this with s instead of s_values?  Would it copy the indices tensor?

Reply:
I think it would yeah. I think it'd also be a bit messier converting src directly since below still needs the values/indices separately, and src is a reference, so there'd be an additional temporary variable that isn't really needed.